File: Assets/Scripts/Farmer.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;


//every farmer has a field and a startValue ''the first corn'' but in kilograms
public class Farmer : MonoBehaviour
{
    private Field field; //his own field
    private List<Farmer> collabFarmer = new List<Farmer>();
    public string name;
    public Color color;
    public Color nameColor;

    public Farmer(Field field, int startValue, string name, Color color)
    {
        this.field = field;
        this.field.SetStartValue(startValue);
        this.name = name;
        this.color = color;
    }

    public List<Farmer> GetCollabFarmer()
    {
        return collabFarmer;
    }
    public Field GetField()
    {
        return field;
    }
    public void SetCollabFarmer(Farmer farmer)
    {
        collabFarmer.Add(farmer);
    }

    public void SetNameColor(Color color)
    {
        nameColor = color;
    }

    public bool HasNoCollabFarmer()
    {
        return collabFarmer.Count == 0;
    }
}

File: Assets/PythonScripts/Farmer.py
# -*- coding: utf-8 -*-
"""
Created on Sun Nov 10 14:10:24 2019

@author: sakob
"""
import Field

class Farmer():
    __field = Field.Field(0) #his own field
    __collabFarmer = None
    name = None

    def __init__(self, field, startValue, name):
        self.__field = field
        self.__field.SetStartValue(startValue)
        self.__collabFarmer = None
        self.name = name
    

    def GetCollabFarmer(self):
        return self.__collabFarmer
    
    def GetField(self):  
        return self.__field
    
    def SetCollabFarmer(self, farmer):
        self.__collabFarmer = farmer
    
    def HasNoCollabFarmer(self):
        return bool(self.__collabFarmer is None)